src/gpio.rs

```diff
@@ -44,6 +53,106 @@ pub struct Output<MODE> {
 /// Push pull output (type state)
 pub struct PushPull;
 
+/// Fully erased pin
+pub struct Pin<MODE> {
+    i: u8,
+    port: *const dyn GpioRegExt,
+    _mode: PhantomData<MODE>,
+}
+
+macro_rules! gpio_trait {
+    ($gpiox:ident) => {
+        impl GpioRegExt for crate::stm32::$gpiox::RegisterBlock {
+            fn is_low(&self, pos: u8) -> bool {
+                // NOTE(unsafe) atomic read with no side effects
+                self.idr.read().bits() & (1 << pos) == 0
+            }
+
+            fn is_set_low(&self, pos: u8) -> bool {
+                // NOTE(unsafe) atomic read with no side effects
+                self.odr.read().bits() & (1 << pos) == 0
+            }
+
+            fn set_high(&self, pos: u8) {
+                // NOTE(unsafe) atomic write to a stateless register
+                unsafe { self.bsrr.write(|w| w.bits(1 << pos)) }
+            }
+
+            fn set_low(&self, pos: u8) {
+                // NOTE(unsafe) atomic write to a stateless register
+                unsafe { self.bsrr.write(|w| w.bits(1 << (pos + 16))) }
+            }
+        }
+    };
+}
+
+gpio_trait!(gpioa);
+gpio_trait!(gpiob);
```

Comment:
Should there be also `gpio_trait!(gpioc)`?

Reply:
And stm32g081RBIx also has `D` port.

Reply:
I checked this: https://github.com/stm32-rs/stm32-rs/blob/master/svd/vendor/en.stm32g0_svd.zip

There is:

```xml
<peripheral derivedFrom="GPIOB">
      <name>GPIOC</name>
      <baseAddress>0x50000800</baseAddress>
    </peripheral>
    <peripheral derivedFrom="GPIOB">
      <name>GPIOD</name>
      <baseAddress>0x50000C00</baseAddress>
    </peripheral>
    <peripheral derivedFrom="GPIOB">
      <name>GPIOF</name>
      <baseAddress>0x50001400</baseAddress>
    </peripheral>
```

I guess those are not handled properly and therefore `CDF` are not generated.

Reply:
Aha! It's just that there are two (otherwise identical) [named](https://docs.rs/stm32g0/latest/src/stm32g0/stm32g081/gpioa.rs.html) [modules](https://docs.rs/stm32g0/latest/src/stm32g0/stm32g081/gpiob.rs.html) for defining the registers for all GPIO ports. The actual ports GPIOA–GPIOF are [fully defined](https://docs.rs/stm32g0/latest/src/stm32g0/stm32g081/mod.rs.html#471) elsewhere. I have no idea why ST decided that with the same registers on A-F, they needed to define it as A and B-F. They could be merged in a patch for stm32-rs, but not sure if that would break anything.

In any case, the answer for this PR is simply that it's only necessary to add GpioRegExt to those two modules to correctly support all six ports, A–F.

I did a quick sanity check downgrading PC6 on a STMG031 and everything seems to be working.